charts/fluentd/templates/_helpers.tpl

```diff
@@ -90,3 +90,12 @@ Name of the configMap used for additional configuration files; allows users to o
     {{ printf "%s-%s" "fluentd-config" ( include "fluentd.shortReleaseName" . ) }}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Fluentd image with tag/digest
+*/}}
+{{- define "fluentd.image" -}}
+{{- $tag := ternary "" (printf ":%s" (toString .tag)) (or (empty .tag) (eq "-" (toString .tag))) -}}
```

Comment:
tags and hashes can coexist, where hash will always takes precedence over the tag, even if they dont match.

With this info you should be able to simplify this. Although it works as it is...